cfme-performance/tests/workloads/test_provisioning.py

```diff
@@ -0,0 +1,116 @@
+from utils.workloads import get_provisioning_scenarios
+from itertools import cycle
+import time
+import pytest
+
+
+@pytest.mark.parametrize('scenario', get_provisioning_scenarios())
+def test_provisioning(request, scenario):
+    """Runs through provisioning scenarios using the REST API to
+    continously provision a VM for a specified period of time.
+    Memory Monitor creates graphs and summary at the end of each scenario."""
+
+    from_ts = int(time.time() * 1000)
+    ssh_client = SSHClient()
+    logger.debug('Scenario: {}'.format(scenario['name']))
+
+    clean_appliance(ssh_client)
+
+    scenario_data = {'appliance_ip': cfme_performance['appliance']['ip_address'],
+        'appliance_name': cfme_performance['appliance']['appliance_name'],
+        'test_dir': 'workload-provisioning',
+        'test_name': 'Test Provisioning',
+        'appliance_roles': get_server_roles_workload_provisioning(separator=', '),
+        'scenario': {'name': 'default'}}
+    monitor_thread = SmemMemoryMonitor(SSHClient(), scenario_data)
+
+    provision_order = []
+
+    def cleanup_workload(scenario, from_ts, vms_to_cleanup):
+        starttime = time.time()
+        to_ts = int(starttime * 1000)
+        g_url = get_scenario_dashboard_url(scenario, from_ts, to_ts)
+        logger.debug('Started cleaning up monitoring thread.')
+        monitor_thread.grafana_url = g_url
+        monitor_thread.signal = False
+        monitor_thread.join()
+        delete_provisioned_vms(vms_to_cleanup)
+        timediff = time.time() - starttime
+        logger.info('Finished cleaning up monitoring thread in {}'.format(timediff))
+    request.addfinalizer(lambda: cleanup_workload(scenario, from_ts, provision_order))
+
+    monitor_thread.start()
+
+    wait_for_miq_server_ready(poll_interval=2)
+    set_server_roles_workload_provisioning(ssh_client)
+    add_providers(scenario['providers'])
+    logger.info('Sleeping for Refresh: {}s'.format(scenario['refresh_sleep_time']))
+    time.sleep(scenario['refresh_sleep_time'])
+
+    guid_list = get_template_guids(scenario['templates'])
+    guid_cycle = cycle(guid_list)
+    providers_num = len(scenario['providers'])
+    cleanup_size = scenario['cleanup_size']
+    total_time = scenario['total_time']
+    time_between_provision = scenario['time_between_provision']
+    total_provisioned_vms = 0
+    total_deleted_vms = 0
+    provisioned_vms = 0
+    starttime = time.time()
+
+    while ((time.time() - starttime) < total_time):
+        start_provision_time = time.time()
+
+        total_provisioned_vms += 1
+        provisioned_vms += 1
+        vm_to_provision = '{}-provision-{}'.format(
+            time.strftime('%Y%m%d%H%M%S'), str(total_provisioned_vms).zfill(3))
+        guid_to_provision = next(guid_cycle)
+        provision_order.append([vm_to_provision, guid_to_provision[1]])
+        provision_vm(vm_to_provision, guid_to_provision[0])
+        if provisioned_vms > cleanup_size * providers_num:
```

Comment:
I think it might make readability a bit better if we just eliminate `providers_num` and use `len(scenario['providers'])`